src/libraries/Microsoft.Extensions.Diagnostics/tests/MetricsTests.cs

```diff
@@ -116,69 +116,35 @@ public void MeterDisposeTest()
             IMeterFactory meterFactory = sp.GetRequiredService<IMeterFactory>();
 
             Meter meter = meterFactory.Create("DisposableMeter");
-
             Counter<int> counter = meter.CreateCounter<int>("MyCounter");
-            InstrumentRecorder<int> recorder = new InstrumentRecorder<int>(counter);
+
+            using MeterListener listener = new MeterListener();
+            listener.InstrumentPublished = (instrument, theListener) =>
```

Comment:
This works and I'm fine if you merge it this way, but a simpler option is:

```C#
using MeterListener listener = new MeterListener();
int lastMeasurement = 0;
listener.SetMeasurementEventCallback<int>((inst, measurement, tags, state) => lastMeasurement = measurement);
listener.EnableMeasurementEvents(counter, null);
```
There is no requirement to use publishing events if you already know the counter you want to listen to.

Reply:
I'll fix this in another PR. Thanks!

Reply:
opened the PR https://github.com/dotnet/runtime/pull/87898